p2p/protocol/holepunch/coordination.go

```diff
@@ -49,11 +50,53 @@ func NewHolePunchService(h host.Host, ids *identify.IDService) (*HolePunchServic
 	ctx, cancel := context.WithCancel(context.Background())
 	hs := &HolePunchService{ctx: ctx, ctxCancel: cancel, host: h, ids: ids}
 
+	sub, err := h.EventBus().Subscribe(new(event.EvtNATDeviceTypeChanged))
+	if err != nil {
+		return nil, err
+	}
+
 	h.SetStreamHandler(protocol, hs.handleNewStream)
 	h.Network().Notify((*netNotifiee)(hs))
+
+	hs.refCount.Add(1)
+	go hs.loop(sub)
+
 	return hs, nil
 }
 
+func (hs *HolePunchService) loop(sub event.Subscription) {
+	defer hs.refCount.Done()
+	defer sub.Close()
+
+	for {
+		select {
+		case _, ok := <-sub.Out():
```

Comment:
shouldn't we look at the actual event? why are we ignoring it?

Reply:
Also, where do we set the support meta-variables into the peerstore?
Shouldn't we be doing it here? Or is it handled anywhere?

I think we need a comment if this is correct as is.

Reply:
@vyzo 

1. Our local vars are set in the peerstore in `basic_host.go` during initialisation of the Host & updated in the obs addr manager.

2. Remote vars are set in the peerstore by Identify.

Both changes are included in this PR. I've added a comment.